### PR TITLE
Move GHRSST support closer to driver pattern

### PR DESCRIPTION
**Pull request by twsearle**
# Refactor of GHRSST to new driver pattern

In the spirit of #299, I added a Dataset class to `drivers/ghrsstl4.py` and `color_mapper` support for the drivers.

I am not quite convinced that this is what you had in mind with the settings dictionary in `main.py`, so let me know what you think things ought to be different.

I want to invest some time in migrating the ghrsstl4 tests to pytest and tidying things up a bit more, but I thought I would check the direction I am going in is OK before going further.

## Check list

- [ ] Bump version number to reflect change, edit `forest/__init__.py`

### CODE REVIEW

**Comment by andrewgryan** [comment]
@twsearle this is a very good development. I'm thinking of migrating the unified model file type to a driver too. I'm looking forward to this refactor going in.

**Comment by andrewgryan** [comment]
@twsearle the label is on the `group.label` in `main.py`, it can be passed in to `drivers.get_dataset(driver_name, settings)`

File: forest/main.py [hunk 1]
```diff
@@ -127,7 +127,8 @@ def main(argv=None):
         else:
             # Use dataset interface
             settings = {
-                "pattern": group.pattern
+                "pattern": group.pattern,
+                "color_mapper": color_mapper
```
Comment (andrewgryan) [inline forest/main.py]: `"label": group.label` could go here

Comment (andrewgryan) [reply forest/main.py]: In the future there will be muliple color_mappers and dataset specific settings but for now this is the desired approach. Pass in all the settings needed to configure all `Datasets` it's a step in the right direction

Comment (twsearle) [reply forest/main.py]: yes, OK sounds good I will make that change. I couldn't remember what the group label did so I thought I could let it quietly slip into the night :)

Comment (andrewgryan) [reply forest/main.py]: It's meta-data for the tooltip. I toyed with the idea of using it as a primary navigation key, but now I think that concept needs to be de-coupled, e.g. assign a uid for each dataset on start up

**Comment by andrewgryan** [comment]
@twsearle is this pretty much ready to merge? E.g. does it still plot GHRSST files?

**Comment by twsearle** [comment]
yes, it might be best to merge this now. It passes the tests and GHRSST files are still plotting. Probably best to put further changes in a separate pull request.

**Comment by twsearle** [comment]
But you will need to do the version number updating depending on when you merge it :)

**Comment by andrewgryan** [comment]
I think a refactor doesn't change the release number as there is no different end-user behaviour, only bugs and api changes edit version numbers

**Comment by andrewgryan** [comment]
@twsearle thanks for the refactor! I think now that a pattern is established it is just a case of applying the same steps to the other "file types", gridded_forecast seems like a good candidate since it is a similar piece of code to your ghrsst.py. Also, dataset.time_series_view() and dataset.profile_view() could be future candidates?

**Comment by twsearle** [comment]
Agreed! ghrsst is just a shameless copy of gridded_forecast at the moment, so I can do similar thing there in another short pull request, although it might have to wait til next week for me.

 I am not quite sure on how to handle the views. If I am understanding correctly, you are suggesting that separate `time_series_view` implementations belong in their respective driver module files? I would be happy to give this a go at some point if I get the chance!

**Comment by andrewgryan** [comment]
@twsearle I think the aim is for the main program to call a function if it's available, e.g. if dataset.map_view() can be called then something can be placed on the map. Likewise, dataset.profile_view() if it's implemented means this dataset supports profiles. The idea is to make the main program generic and the dataset responsible for defining it's own representations

**Pull request merged**